src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java

```diff
@@ -67,6 +73,93 @@
+            purchaseOrderNumber = findPurchaseOrderNymberForInvoiceOptionContractControlAndScheduled(document);
+        }
+        
+        if (StringUtils.isBlank(purchaseOrderNumber)) {
+            LOG.error("setPurchaseOrderNumberFieldInParameterMap, unable to find the purchase order number for document " + document.getDocumentNumber());
+        }
+        
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("setPurchaseOrderNumberFieldInParameterMap, for document " + document.getDocumentNumber() + " with an invoice option code of "
+                    + invoiceOptionCode + " the purchase order number is " + purchaseOrderNumber);
+        }
+
+        localParameterMap.put(CuArPropertyConstants.ContractsAndGrantsBillingAwardFields.AWARD_PURCHASE_ORDER_NBR, purchaseOrderNumber);
+    }
+
+    protected String findPurchaseOrderNumberForInvoiceOptionAward(Award award) {
+        String purchaseOrderNumber = StringUtils.EMPTY;
+        AwardExtendedAttribute awardExtension = (AwardExtendedAttribute) award.getExtension();
+        if (ObjectUtils.isNotNull(awardExtension)) {
+            purchaseOrderNumber = awardExtension.getPurchaseOrderNumber();
+        }
+        return purchaseOrderNumber;
+    }
+
+    protected String findPurchaseOrderNumberForInvoiceOptionAccoiunt(ContractsGrantsInvoiceDocument document) {
+        String purchaseOrderNumber = StringUtils.EMPTY;
+        List<InvoiceAccountDetail> invoiceAccountDetails = document.getAccountDetails();
+        if (CollectionUtils.isNotEmpty(invoiceAccountDetails)) {
+            Account invoiceAccount = invoiceAccountDetails.get(0).getAccount();
+            AwardAccountExtendedAttribute awardAccountExtension = findAwardAccountExtendedAttribute(invoiceAccountDetails.get(0).getProposalNumber(),
+                    invoiceAccount.getChartOfAccountsCode(), invoiceAccount.getAccountNumber());
+            if (ObjectUtils.isNotNull(awardAccountExtension)) {
+                purchaseOrderNumber = awardAccountExtension.getAccountPurchaseOrderNumber();
+            }
+        }
+        return purchaseOrderNumber;
+    }
+
+    protected AwardAccountExtendedAttribute findAwardAccountExtendedAttribute(String proposalNumber, String chartOfAccountsCode, String accountNumber) {
+        Map<String, Object> primaryKeys = new HashMap<String, Object>();
+        primaryKeys.put(KFSPropertyConstants.PROPOSAL_NUMBER, proposalNumber);
+        primaryKeys.put(KFSPropertyConstants.CHART_OF_ACCOUNTS_CODE, chartOfAccountsCode);
+        primaryKeys.put(KFSPropertyConstants.ACCOUNT_NUMBER, accountNumber);
+        AwardAccountExtendedAttribute awardAccountExtension = businessObjectService.findByPrimaryKey(AwardAccountExtendedAttribute.class, primaryKeys);
+        return awardAccountExtension;
+    }
+
+    protected String findPurchaseOrderNymberForInvoiceOptionContractControlAndScheduled(ContractsGrantsInvoiceDocument document) {
```

Comment:
Minor typo in method name; should be changed from "...OrderNymber..." to "...OrderNumber..."

Reply:
ok